nibabel/tests/test_spatialimages.py

```diff
@@ -634,28 +631,24 @@ def test_load_mmap(self):
                     back_img = func(param1, **kwargs)
                     back_data = np.asanyarray(back_img.dataobj)
                     if expected_mode is None:
-                        assert_false(isinstance(back_data, np.memmap),
-                                     'Should not be a %s' % img_klass.__name__)
+                        assert not isinstance(back_data, np.memmap), 'Should not be a %s' % img_klass.__name__
                     else:
-                        assert_true(isinstance(back_data, np.memmap),
-                                    'Not a %s' % img_klass.__name__)
+                        assert isinstance(back_data, np.memmap), 'Not a %s' % img_klass.__name__
                         if self.check_mmap_mode:
-                            assert_equal(back_data.mode, expected_mode)
+                            assert back_data.mode == expected_mode
                     del back_img, back_data
                 # Check that mmap is keyword-only
-                assert_raises(TypeError, func, param1, True)
+                with pytest.raises(TypeError):
+                    func(param1, True)
                 # Check invalid values raise error
-                assert_raises(ValueError, func, param1, mmap='rw')
-                assert_raises(ValueError, func, param1, mmap='r+')
+                with pytest.raises(ValueError):
+                    func(param1, mmap='rw')
+                with pytest.raises(ValueError):
+                    func(param1, mmap='r+')
 
 
 def test_header_deprecated():
-    with clear_and_catch_warnings() as w:
-        warnings.simplefilter('always', DeprecationWarning)
-
+    with pytest.deprecated_call():
         class MyHeader(Header):
             pass
-        assert_equal(len(w), 0)
-
-        MyHeader()
-        assert_equal(len(w), 1)
+        MyHeader()
```

Comment:
This test is checking that the `DeprecationWarning` is raised on initialization, not on definition. So we should check *when* it is raised.

```suggestion
        assert len(w) == 0
        MyHeader()
```